apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Footer/HostSidePanel/HostSidePanel.tsx

```diff
@@ -60,22 +89,39 @@ export function HostSidePanel(): ReactElement {
   }
 
   const handleSelectHost = async (hostId: string): Promise<void> => {
-    setSelectedHost(data.hosts.find((host) => host.id === hostId))
+    setSelectedHost(teamHosts.hosts.find((host) => host.id === hostId))
     setOpenSelect(false)
   }
 
   return (
     <>
       {/* DefaultHostPanel - no host */}
       <SidePanel title={t('Hosted By')} open edit>
-        {!openSelect && selectedHost == null && (
-          <SidePanelContainer border={false}>
-            <ContainedIconButton
-              label={t('Select a Host')}
-              thumbnailIcon={<UserProfiles2Icon />}
-              onClick={() => setOpenSelect(true)}
-            />
-          </SidePanelContainer>
+        {!openSelect && (selectedHost == null || !userInTeam) && (
+          <>
+            <SidePanelContainer border={!userInTeam}>
+              <ContainedIconButton
+                label={t('Select a Host')}
+                disabled={!userInTeam}
+                thumbnailIcon={<UserProfiles2Icon />}
+                onClick={() => setOpenSelect(true)}
```

Comment:
Do we want to pass this in if we know its going to be disabled? Would it be safer to do this instead in case we miss some logic check in the future?
```suggestion
                onClick={noop}
```

Reply:
We only use `noop` for tests, not for normal components. Still need to keep the onClick behaviour, it just won't be triggered if the button is disabled.